debian/snapcraft-completion

```diff
@@ -6,15 +20,15 @@ _snapcraft()
     COMPREPLY=()
     cur="${COMP_WORDS[COMP_CWORD]}"
     prev="${COMP_WORDS[COMP_CWORD-1]}"
-    opts="help init list-plugins plugins login logout export-login list-keys keys create-key register-key register registered list-registered push release clean cleanbuild pull build sign-build stage prime snap update define search gated validate history status close enable-ci expand-extensions extension extensions list-extensions"
+    opts="help init list-plugins plugins login logout export-login list-keys keys create-key register-key register registered list-registered push release clean cleanbuild pull build sign-build stage prime pack update define search gated validate history status close enable-ci expand-extensions extension extensions list-extensions --target-arch --build-for --debug --shell --shell-after ---provider -h --help --verbose"
```

Comment:
Most of these new parameters should only be available if a lifecycle command has already been declared. For example, `snapcraft plugins --build-for` isn't valid

Reply:
@lengau that is fair point, let me see if this can be fixed in easy way

Reply:
Probably the easiest way would be to check whether any of the lifecycle commands are in `$COMP_WORDS`:

```
for i in "${COMP_WORDSS}"; do
 case "$i" in  
  pull);&
  build);&
  stage);&
  prime);&
  pack);& 
  clean) 
    opts="${opts} --build-for --target-arch --debug --shell --shell-after --provider"
    break
 esac
done```

Reply:
@kubiko any update?